### DISCUSSION TITLE
livewire 3 url id remove after $this->reset

### DISCUSSION THREAD

**Discussion by dharen008** [livewire/livewire#7782] [1. Help]
If you use $this->reset() in a Livewire component, it will reset all the data, including $dataId, to their initial values. If you want to keep $dataId unchanged when using $this->reset(), you currently need to specify all the other fields explicitly like $this->reset('name1', 'name2', 'etc'). Is there a way to prevent $dataId from being reset without listing all the fields individually?

In both examples, the $dataId gets reset.

Route::get( '/employee/{dataId}/pto', PtoDetails::class );

http://domain.com/employee/7777777/pto

Example 1:
`public $dataId;
public $thisUser;

public function mount() {
if ( $this->dataId ) {
$this->thisUser = User::find( $this->dataId );
}
}`

Example 2:
` public $thisUser;

public function mount( Users $dataId ) {
if ( $dataId ) {
$this->thisUser = $dataId;
}
}`

**Comment by hapidjus** [comment]
You should use `$this->resetExcept('dataId')` if you want to reset all other fields.
